Clone Detection Evaluator

kritipat opened the pull request:
I have added the evaluator script for the clone detector task. The README should be updated to contain information on what the input and output is and how to run it.

Original: https://github.com/microsoft/CodeXGLUE/tree/main/Code-Code/Clone-detection-BigCloneBench/evaluator

kritipat committed:
Update eval for clone detection

kritipat committed:
Update README.md

kritipat committed:
Update README.md

mrhuggins03 commented:
Hi @kritipat , 

Thank you for your contribution! 

For the input and output section of your readme, I think the format is incorrect. It seems to me that the format for answers should be like this file [here](https://github.com/microsoft/CodeXGLUE/blob/main/Code-Code/Clone-detection-BigCloneBench/evaluator/answers.txt). The predictions file would be similar. I could very well be missing something though.

I also see that you added the How To Test section but we are also requiring that you add a test file. You will find a starter file for the testing script in the test folder under your task. The script will test that the output of the evaluator is correct. You can find an example [here](https://github.com/SEEDGuard/SEVAL/blob/main/src/test/code_generation/test.py). This script will use the answers and predictions files from the CodeXGLUE repository.

If you have any questions please let me know.

kritipat committed:
Add files via upload

kritipat committed:
Added prediction file

kritipat committed:
Corrected input and output file formats

kritipat committed:
Update README.md

kritipat committed:
Added code to test.py

kritipat commented:
Thank you for the feedback. I have updated the README and added appropriate files to for the tasks and test folder. Could you please review @mrhuggins03?

mrhuggins03 commented:
Hey @kritipat, 

I tried to run your script but you reference "expected.txt" and I do not see that file in your changes. Also you use a variable called total_lines that I do not see defined anywhere.

You do not need to calculate the number of correct lines. The evaluation script you added earlier should print the accuracy and the test script you are writing is checking that the output is correct. You can find the expected output in the readme of your task in the CodeXGLUE repo.

kritipat committed:
Fixed README

kritipat committed:
Update README.md
